Adding a contributor guide in a separate file

gkunz opened the pull request:
Moving the contribution instructions from the README to a dedicated CONTRIBUTING.md file, following best practices.

gkunz committed:
Adding a contributor guide in a separate file

Moving the contribution instructions from the README to a dedicated
CONTRIBUTING.md file, following best practices.

tomas-ostling-lt commented on CONTRIBUTING.md:
I think the Contribution routine is good, and will work fine for anyone who is used to the Github lingo. However, it is proven during the work, that not all developers (and even more so contributors in secondary target groups, e.g. (roject managers, line managers) are not usually working in Github. 
I approve with posting the guide in a separate file, but suggest it is given some work  to make it easier to understand for non Github users.

YasinAhmed2 replied:
I agree the Contribution guide looks good to me too. Clear, short and straight forward. May be we could add a little about Code of Conduct?

tomas-ostling-lt reviewed (commented):
I think the Contribution routine is good, but  not all developers (and even more so contributors in secondary target groups (e.g. project managers, line managers) are not usually working in Github, and therefore I suggest both CONTRIUTION.md and README.md is given some work to make it easier to understand for non Github users. 

The collaboration routine is also good, but will need to be compared to the content of the PECA content update routine (PECA-Content-update-routine2023.pptx).  

The changes about ELL will need to be reviewed by a partner representative.  
Therefor, I must leave this contribution without approval...

gkunz commented:
Hi @tomas-ostling-lt, @Ramses-FOSS, @MaArad 

thanks for the feedback. Some comments:

i) extending the contribution guide: Agreed that the current guide is very concise (in fact, it is only a move of existing content). However, I would not add a lot of additional material to explain to non-GitHub users how GitHub works, but instead point to the official GitHub documentation, which is excellent and which will be kept up-to-date by GitHub itself [1]. No need to duplicate this.

ii) relationship to PECA update procedure: Yes, we should add that fundamental structural updates of the material need to go through a more involved process. However, this is the contribution guide for the GitHub repo only. If we want to keep the current exercise of asking for a example PR, we cannot make the contribution guide point to a different procedure. Probably we should create a dedicated sample repository for training-PRs submitted by users, i.e., not point them to this repo here.

iii) Ok. However, the sentence I deleted was wrong anyway in the sense that it said that in the future the material will be moved to the ELL repo. This here IS the ELL repo, so this sentence does not make sense.

[1] https://docs.github.com/en/get-started/using-github/github-flow

gkunz committed:
Extending the contribution guide and adding a file for the contribution exercise

Adding more details to the contribution guide, differentiating between
contributions to improve the training material itself and the
contribution exercise.

gkunz commented:
@tomas-ostling-lt: I have updated the contribution guide, including a comment on the peca player, and differentiating between the contribution exercise and modifications of the material itself. For details on the GitHub lingo and workflow, I refer to the GitHub documentation.

gkunz committed:
Adding an incorrectly spelled word to the contribution exercise

YasinAhmed2 commented on Contributions-exercise.md:
```diff
@@ -0,0 +1,3 @@
+# Contribution Exercise
+
+Make changes to this file as part of the contribution exercise. Feel free to add markdown content with correct or incorrect syntax and observe the results of the markdown linter. Do you find the wrod with the wrong spelling and can you fix it?
```
If several persons are taking the coarse at the same time, How will this work? If the first person has already made the changes (for instance found wrod, changes it word) and the PR request is approved, what would the next person practice on? May be we should encourage to just feed the next incorrect syntax, if it is not there?

gkunz replied:
This typo is not meant to be corrected. The contribution guide states that such PRs will not get merged, thus the typo remains for everybody to propose a fix for: https://github.com/ExpertLearningLab/foss-learning/blob/main/CONTRIBUTING.md?plain=1#L32

YasinAhmed2 replied:
Great, thanks for the clarification

YasinAhmed2 reviewed (commented):
Looks good with Contribution Guide and Exercise, with some comments. No comments on ReadMe file, where I expect the Expertlearning to describe the final version as a description for the Repo.
